compiler/circle-mpqsolver/src/CircleMPQSolver.cpp

```diff
@@ -74,6 +89,78 @@ int entry(int argc, char **argv)
     setenv("LUCI_LOG", "100", 0);
   }
 
-  // TODO Run Bisect method
+  auto data_path = arser.get<std::string>("--data");
+  auto input_model_path = arser.get<std::string>("--input_model");
+  auto output_model_path = arser.get<std::string>("--output_model");
+  auto input_quantization = arser.get<std::string>("--input_dtype");
+  auto output_quantization = arser.get<std::string>("--output_dtype");
+
+  float qerror_ratio = arser.get<float>("--qerror_ratio");
+  if (qerror_ratio < 0.0 || qerror_ratio > 1.f)
+  {
+    std::cerr << "ERROR: quantization ratio must be in [0, 1]" << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto start = std::chrono::high_resolution_clock::now();
+
+  if (arser[bisection_str])
+  {
+    // optimize
+    using namespace mpqsolver::bisection;
+
+    BisectionSolver solver(data_path, qerror_ratio, input_quantization, output_quantization);
+    {
+      auto value = arser.get<std::string>(bisection_str);
+      if (value == "auto")
+      {
+        // TODO
+        std::cerr << "Not implemented yet " << input_model_path << std::endl;
+        return EXIT_FAILURE;
+      }
+      else if (value == "true")
+      {
+        solver.algorithm(BisectionSolver::Algorithm::ForceQ16Front);
+      }
+      else if (value == "false")
+      {
+        solver.algorithm(BisectionSolver::Algorithm::ForceQ16Back);
+      }
+      else
+      {
+        std::cerr << "Unrecognized option for bisection algortithm" << input_model_path
+                  << std::endl;
+        return EXIT_FAILURE;
+      }
+    }
+
+    auto optimized = solver.run(input_model_path);
+    if (optimized == nullptr)
+    {
+      std::cerr << "Failed to build mixed precision model" << input_model_path << std::endl;
+      return EXIT_FAILURE;
+    }
+
+    // save optimized
+    {
+      luci::CircleExporter exporter;
+      luci::CircleFileExpContract contract(optimized.get(), output_model_path);
+      if (!exporter.invoke(&contract))
+      {
+        std::cerr << "Failed to export mixed precision model" << input_model_path << std::endl;
+        return EXIT_FAILURE;
+      }
+    }
+  }
+  else
+  {
+    std::cerr << "Unrecognized solver" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  auto duration = std::chrono::duration_cast<std::chrono::seconds>(
+    std::chrono::high_resolution_clock::now() - start);
+  std::cerr << "Elapsed Time: " << std::setprecision(5) << duration.count() / 60.f << " minutes."
```

Comment:
This doesn't seem to be an error. If not, can we use `std::cout` ?

Reply:
If possible can we use `LOGGER` ?

Reply:
ok